app/gui2/src/stores/project/executionContext.ts

```diff
@@ -274,17 +274,22 @@ export class ExecutionContext extends ObservableV2<ExecutionContextNotification>
   }
 
   dispose() {
+    const abortController = new AbortController()
+    this.lsRpc.willRelease(abortController)
```

Comment:
`willRelease` is very misleading: I would expect something like "will release when aborted", but this actually aborts on _disconnect_

Also, I think `abortController` is not needed here at all; we just release LS once connection is closed.

Reply:
Also, I think it should be handled on higher level: in the future we might have more "cleanups" in LsRpc. So instead `willRelease` I would make method `stopReconnecting` and call it not here, but somewhere "higher" at the moment when we're sure the GUI app is tearing down.

Reply:
that sounds like it would make `retainCount`/`release`/`retain` redundant, right? since now `stopReconnecting` is what determines whether the LS transport/connection is disposed

Reply:
No, because you may close the GUI App without closing LS - in that case LS never closes connection